modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb

```diff
@@ -475,12 +475,18 @@ def embedded_data
         end
 
         def embedded_data_match(key)
-          match = appointment.dig(:reason_code, :text)&.match(/(^|\|)#{key}:?(.*?)(\||$)/)
+          camelized_key = key.gsub(' ', '_').camelize(:lower)
```

Comment:
So you're telling me that they're sometimes giving us keys like `preferred dates` and sometimes `preferredDates`? That seems ridiculously sloppy, like they're setting us up for failure.

Reply:
I've only seen this happen with the "reason code" vs reasonCode but I figured it wouldn't be impossible for it to happen to other fields as well so wanted to code defensively for it.

Reply:
Does this PR fix the reason code issue? That's not apparent to me.

Reply:
Yeah, it goes through all the embedded data fields, one of which is the reason code and checks for normal spaced case or camel case as a fall back. so no change in behavior if the original field is there.

Reply:
I see. So when you say reason_code, you mean:
`reason_code[:text][:reason_code]`
not the top level reason code.